tee-worker/litentry/core/common/src/web3_token/mod.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry. If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+use litentry_primitives::Web3TokenType;
+
+use crate::Web3Network;
+
+pub trait TokenName {
+	fn get_token_name(&self) -> &'static str;
+}
+
+impl TokenName for Web3TokenType {
+	fn get_token_name(&self) -> &'static str {
+		match self {
+			Self::Bnb => "BNB",
+			Self::Eth => "ETH",
+			Self::SpaceId => "SPACE_ID",
+			Self::Lit => "LIT",
+			Self::Wbtc => "WBTC",
+			Self::Usdc => "USDC",
+			Self::Usdt => "USDT",
+			Self::Crv => "CRV",
+			Self::Matic => "MATIC",
+			Self::Dydx => "DYDX",
+			Self::Amp => "AMP",
+			Self::Cvx => "CVX",
+			Self::Tusd => "TUSD",
+			Self::Usdd => "USDD",
+			Self::Gusd => "GUSD",
+			Self::Link => "LINK",
+			Self::Grt => "GRT",
+			Self::Comp => "COMP",
+			Self::People => "PEOPLE",
+			Self::Gtc => "GTC",
+			Self::Ton => "TON",
+			Self::Trx => "TRX",
+		}
+	}
+}
+
+pub trait TokenAddress {
+	fn get_token_address(&self, network: Web3Network) -> Option<&'static str>;
+}
+
+impl TokenAddress for Web3TokenType {
+	fn get_token_address(&self, network: Web3Network) -> Option<&'static str> {
+		match (self, network) {
+			// Bnb
+			(Self::Bnb, Web3Network::Ethereum) =>
```

Comment:
Maybe could be a different issue, but maybe we could use a `derive` macro to generate this code instead of having to manually write everything, wdyt?

Reply:
I felt below derive macro example would not be too much simpler than the code above, and less flexible too, what do you think?
 
```
#[derive(TokenAddress)]
pub enum Web3TokenType {
    #[token_address(network = Web3Network::Ethereum => "0xb8c77482e45f1f44de1745f52c74426c631bdd52")]
    #[token_address(network = Web3Network::Bsc => "0x2170ed0880ac9a755fd29b2688956bd959f933f8")]
    Bnb,
    // Add more token types as needed
}
```

Reply:
I'm talking something more like, makes it more generic and easier to add more tokens
```
#[token[Ethereum(0xdeeadbeef), Bsc(0xdeadbeef)]]
Lit
```